mozilla-services/consvc-shepherd
# [SHEP-56] Set up jest testing and typescript eslint for the ad-ops-dashboard

## References

[SHEP-56](https://mozilla-hub.atlassian.net/browse/SHEP-56)

## Problem Statement

We need to set up linting and testing for the new React dashboard. 

## Proposed Changes

In this change, we
- Update the `Makefile` to include testing and linting the dashboard when running the usual `make local-test`, `make test` and `make lint` commands
- Use python-node images in CircleCI so that we can get node/npm out of the box for those builds
- Configure eslint for typescript and vite
- Configure jest testing and dashboard code coverage (though leaving requirement of coverage commented out for now)
- Add a first-pass jest test
- Clean up the products-fetching code 

## Verification Steps

Please run: 
- `make local-test` locally and verify both the shepherd and dashboard test suites are running fully and correctly
- `make eslint` and `make eslint-fix` locally and verify that dashboard is being linted properly. Left the `eslint` command out of the `make lint` command for now since we have some additional lint fixes that need to go in before this passes, and this PR is large enough
- Check the CI output for the latest, successful runs for the `lint` and `test` steps and verify the same as above

[SHEP-56]: https://mozilla-hub.atlassian.net/browse/SHEP-56?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
